orchagent/intfsorch.cpp

```diff
@@ -160,7 +160,12 @@ bool IntfsOrch::setIntf(const string& alias, sai_object_id_t vrf_id, const IpPre
 
     if (port.m_type == Port::VLAN && ip_prefix->isV4())
     {
-        addDirectedBroadcast(port, ip_prefix->getBroadcastIp());
+        /* For /31 IPv4 subnets, there is no broadcast address, hence don't
+         * add a net directed broadcast route specific neighbor entry */
+        if (ip_prefix->getMaskLength() != 31)
+        {
+            addDirectedBroadcast(port, ip_prefix->getBroadcastIp());
```

Comment:
We need to fix getBroadcastIp() and have a check inside addDirectedBroadcast(). We shouldn't be explicitly checking for mask lengths in the code. We just need the APIs to do the right thing.

Reply:
I agree in general of APIs doing the right thing. Pushing the /31 checking logic into addDirectedBroadcast(). The getBroadcastIp() just returns broadcast IP irrespective of subnet size. So, i don't think it needs to be changed here.

Reply:
Changed with review comments incorporated are submitted.